### ISSUE TITLE
Addon fails on start - error: could not get iptables version

### ISSUE THREAD

**Issue by Mincka** [hassio-addons/addon-tailscale#20]
# Problem/Motivation

Addon cannot start properly on Home Assistant OS (HA Blue on Odroid N2+).
Follow up of #12 after merge of #17

## Actual behavior
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: Tailscale
 Zero config VPN for building secure networks
-----------------------------------------------------------
 Add-on version: 0.2.0
 You are running the latest version of this add-on.
 System: Home Assistant OS 6.2  (aarch64 / odroid-n2)
 Home Assistant Core: 2021.8.8
 Home Assistant Supervisor: 2021.06.8
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] nginx.sh: executing... 
[cont-init.d] nginx.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[20:56:33] INFO: Starting Tailscale...
[20:56:33] INFO: Starting Tailscale web...
2021/08/24 20:56:33 logtail started
2021/08/24 20:56:33 Program starting: v1.14.0-t5cff36945-g809e87bba, Go 1.17-tsdd6b269c77: []string{"/opt/tailscaled", "--state=/data/tailscaled.state"}
2021/08/24 20:56:33 LogID: 1ab4cd9fc11bcc69850e073b6270286a5d90a9fc75342bc83230d411177c5abc
2021/08/24 20:56:33 logpolicy: using system state directory "/var/lib/tailscale"
logpolicy.Read /var/lib/tailscale/tailscaled.log.conf: open /var/lib/tailscale/tailscaled.log.conf: no such file or directory
2021/08/24 20:56:33 web server running on: http://127.0.0.1:25899
2021/08/24 20:56:33 wgengine.NewUserspaceEngine(tun "tailscale0") ...
2021/08/24 20:56:33 wgengine.NewUserspaceEngine(tun "tailscale0") error: could not get iptables version: exit status 1
2021/08/24 20:56:33 wgengine.New: could not get iptables version: exit status 1
2021/08/24 20:56:33 flushing log.
2021/08/24 20:56:33 logger closing down
2021/08/24 20:56:33 logtail: dialed "log.tailscale.io:443" in 186ms
[20:56:34] INFO: Starting NGinx...
[cont-finish.d] executing container finish scripts...
[cont-finish.d] 99-message.sh: executing... 
[cont-finish.d] 99-message.sh: exited 0.
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
```

**Comment by RampedIndent** [comment]
My addon starts fine however its not forwarding ip tables. so i cannot use Subnet routes

![image](https://user-images.githubusercontent.com/39007301/131312676-5e58b92b-cea6-4a11-88bf-f7280ef9a8db.png)

```
2021/08/30 18:46:56 router: disabling tunneled IPv6 due to system IPv6 config: kernel doesn't support IPv6 policy routing: ip -6 rule failed: RTNETLINK answers: Address family not supported by protocol
2021/08/30 18:46:56 dns: [rc=unknown ret=dns.directManager]
2021/08/30 18:46:56 dns: using dns.directManager
2021/08/30 18:46:56 link state: interfaces.State{defaultRoute=eth0 ifs={docker0:[172.17.0.1/16] eth0:[192.168.50.24/24] hassio:[172.30.32.1/23]} v4=true v6=false}
2021/08/30 18:46:56 Creating wireguard device...
2021/08/30 18:46:56 Bringing wireguard device up...
2021/08/30 18:46:56 Bringing router up...
2021/08/30 18:46:56 external route: up
2021/08/30 18:46:56 router: failed to delete legacy rule, continuing anyway: checking for [-m comment --comment tailscale -i tailscale0 -j ACCEPT] in filter/FORWARD: running [/sbin/iptables -t filter -C FORWARD -m comment --comment tailscale -i tailscale0 -j ACCEPT --wait]: exit status 2: iptables v1.8.7 (nf_tables): Couldn't load match `comment':No such file or directory
Try `iptables -h' or 'iptables --help' for more information.
2021/08/30 18:46:56 router: failed to delete legacy rule, continuing anyway: checking for [-m comment --comment tailscale -o eth0 -j MASQUERADE] in nat/POSTROUTING: running [/sbin/iptables -t nat -C POSTROUTING -m comment --comment tailscale -o eth0 -j MASQUERADE --wait]: exit status 2: iptables v1.8.7 (nf_tables): Couldn't load match `comment':No such file or directory
Try `iptables -h' or 'iptables --help' for more information.
2021/08/30 18:46:56 Clearing router settings...
2021/08/30 18:46:56 Starting link monitor...
2021/08/30 18:46:56 Starting magicsock...
2021/08/30 18:46:56 Engine created.
```

**Comment by RampedIndent** [comment]
Unsure if this is relevent however when poking around i determind a different version was installed on my ha install compared to the docker file
![image](https://user-images.githubusercontent.com/39007301/131317186-1eadd351-1779-477a-9936-ebd86ed423a9.png)

https://github.com/hassio-addons/addon-tailscale/blob/f7550a0330134117c1c7c76578db6aa7b8f0ffec/tailscale/Dockerfile#L15

**Comment by Steaders1965** [comment]
I'm running HA OS 6.2, Supervisor 2021.09.0 and core-2021.9.4 on an ODROID C2 and I'm having the same problem. Tailscale starts and then immediately stops by sending TERM and KILL sigs to all processes

**Comment by vandenbogerd** [comment]
Same here…
HA Blue with Home Assistant OS 6.3, supervisor-2021.09.0, core-2021.9.4

**Comment by dnedeltchev** [comment]
I thought I was the only with this issue. I have the same issue as @vandenbogerd 
Although I am on core-2021.9.5 and about to upgrade to core-2021.9.6 but I don't think this makes a difference.

**Comment by dnedeltchev** [comment]
> I thought I was the only with this issue. I have the same issue as @vandenbogerd
> Although I am on core-2021.9.5 and about to upgrade to core-2021.9.6 but I don't think this makes a difference.

As I thought, core-2021.9.6 didn't help.

**Comment by andyjsawyer** [comment]
Same here HA Blueseemed to happen after the update of the addon.waiting to see if another update will alter the issue

**Comment by tarsnake** [comment]
Same issue and log output, Odroid C2/HA Blue, release 2021.9.7

**Comment by DunklerPhoenix** [comment]
Same here with HA Blue 2021.9.7

**Comment by pemontto** [comment]
Possibly needs to use legacy iptables?

```dockerfile
RUN update-alternatives --set iptables /usr/sbin/iptables-legacy
RUN update-alternatives --set ip6tables /usr/sbin/ip6tables-legacy
```

**Comment by mostlychris** [comment]
Having the same issue. No solution to offfer atm.

**Comment by Mincka** [comment]
For anyone interested, there is another "non-official" addon that works great on Home Assistant OS, including access to local subnet and usage of HA as exit node.
https://github.com/tsujamin/hass-addons/tree/main/tailscale
The trick is to enable the `userspace_networking` option in the addon configuration.
More info here: https://github.com/hassio-addons/addon-tailscale/issues/22

Maybe it could have been the "proper" way to close #12 before #17 that caused more harm than the original version.

**Comment by frenck** [comment]
Fixed in v0.3.0

Tested on a Home Assistant Blue and a Supervised installation.
